### PR TITLE
Usermanual: add high-level material

### PR DESCRIPTION
**Pull request by n8willis**
This updates the opening few chapters of the user manual. The most significant change is the addition of a 'Text shaping concepts' chapter. It also adds (shorter) explanatory material to the 'What is HarfBuzz' chapter to serve as better first-read content and updates the installation and build instructions.

### CODE REVIEW

**Comment by khaledhosny** [comment]
While at it, you might want to address this issue as well https://github.com/harfbuzz/harfbuzz/issues/298.

File: docs/usermanual-getting-started.xml [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+      optional font features applied.
+    </para>
+    <para>
+      In addition to holding the pre-shaping input (the Unicode
+      codepoints that comprise the input string) and the post-shaping
+      output (the glyphs and positions), a HarfBuzz buffer has several
+      properties that affect shaping. The most important are the
+      text-flow direction (e.g., left-to-right, right-to-left,
+      top-to-bottom, or bottom-to-top), the script tag, and the
+      language tag. HarfBuzz can attempt to guess the correct values
```
[the remark this answers is not in the document]
Comment (behdad) [reply docs/usermanual-getting-started.xml]: Agreed.

File: docs/usermanual-getting-started.xml [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+      explicitly.
+    </para>
+
+    <para>
+      For input string buffers, flags are available to denote when the
+      buffer represents the beginning or end of a paragraph, to
+      indicate whether or not to visibly render Unicode <literal>Default
+      Ignorable</literal> codepoints, and to modify the cluster-merging
+      behavior for the buffer. For shaped output buffers, the
+      individual X and Y offsets and widths of each glyph are
```
[the remark this answers is not in the document]
Comment (n8willis) [reply docs/usermanual-getting-started.xml]: Yeah, I was attempting to not rely on specialist terminology in this section where avoidable. Knowing what an 'advance' is requires background. May be better to say it both ways....

File: docs/harfbuzz-docs.xml [hunk 3]
```diff
@@ -12,37 +12,29 @@
...
-        Also available on
-        <ulink url="https://github.com/harfbuzz/harfbuzz">github</ulink>.
+	The canonical source-code tree is available at
+        <ulink
+	    url="http://cgit.freedesktop.org/harfbuzz/">cgit.freedesktop.org/harfbuzz</ulink>
+	and at
+        <ulink
+	    url="https://github.com/harfbuzz/harfbuzz">github.com/harfbuzz/harfbuzz</ulink>. 
         See <xref linkend="download" endterm="download.title"/> for release tarballs.
-      </para>
```
[the remark this answers is not in the document]
Comment (n8willis) [reply docs/harfbuzz-docs.xml]: Hmm. Not sure why that wasn't caught by Emacs. More importantly, though, fixing it doesn't seem to resolve the build failures.

File: docs/usermanual-install-harfbuzz.xml [hunk 4]
```diff
@@ -1,70 +1,409 @@
...
+	
+	<varlistentry>
+	  <term>--with-uniscribe</term>
+	  <listitem>
+	    <para>
+	      Use the <ulink url="https://docs.microsoft.com/en-us/windows/desktop/intl/uniscribe">Uniscribe</ulink> library. <emphasis>(Default = no)</emphasis>
+	    </para>	    
+	    <para>
+	      This option enables or disables usage of the Uniscribe
+	      font-rendering library. Uniscribe is available on Windows systems.
```
[the remark this answers is not in the document]
Comment (n8willis) [reply docs/usermanual-install-harfbuzz.xml]: How do you mean?

**Comment by behdad** [comment]
I haven't read this, but is this mergeable?  I haven't read the user manual that Simon contributed.  I rather we merge and then I review it all and suggest.

File: docs/usermanual-what-is-harfbuzz.xml [hunk 5]
```diff
@@ -1,115 +1,433 @@
...
+      In addition to OpenType shaping, HarfBuzz supports the latest
+      version of Graphite shaping. HarfBuzz currently supports AAT
+      shaping only on macOS and iOS systems.
+    </para>
+    
+    <para>
+      HarfBuzz can read and understand TrueType fonts (.ttf), TrueType
+      collections (.ttc), and OpenType fonts (.otf, including those
+      fonts that contain TrueType-style outlines and those that
+      contain PostScript CFF or CFF2 outlines).
```
Comment (ebraminio) [inline docs/usermanual-what-is-harfbuzz.xml]: And dfont, if someone care :)

File: docs/usermanual-what-is-harfbuzz.xml [hunk 6]
```diff
@@ -1,115 +1,433 @@
...
+	  Emoji (including emoji modifier sequences, flag sequences,
+	  and ZWJ sequences)
+	</para>
+      </listitem>
+    </itemizedlist>
+
+    <para>
+      In addition to OpenType shaping, HarfBuzz supports the latest
+      version of Graphite shaping. HarfBuzz currently supports AAT
+      shaping only on macOS and iOS systems.
```
Comment (ebraminio) [inline docs/usermanual-what-is-harfbuzz.xml]: Well, CoreText itself is not available anywhere else :)

File: docs/usermanual-what-is-harfbuzz.xml [hunk 7]
```diff
@@ -1,115 +1,433 @@
...
+      behavior. Implementing these rules is the job of the shaping engine.
+    </para>
+    <para>
+      Text shaping is a fairly low-level operation. HarfBuzz is
+      used directly by text-handling libraries like <ulink
+      url="https://www.pango.org/">Pango</ulink>, as well as by the layout
+      engines in Firefox, LibreOffice, and Chromium. Unless you are
+      <emphasis>writing</emphasis> one of these layout engines
+      yourself, you will probably not need to use HarfBuzz: normally,
+      lower-level libraries will turn text into glyphs for you.
```
Comment (ebraminio) [inline docs/usermanual-what-is-harfbuzz.xml]: Higher maybe :)

File: docs/harfbuzz-docs.xml (hunk 3 above)
[the remark this answers is not in the document]
Comment (khaledhosny) [reply docs/harfbuzz-docs.xml]: You still have more parser errors https://travis-ci.org/harfbuzz/harfbuzz/jobs/444118055#L2552:
```
../usermanual-what-is-harfbuzz.xml:201: parser error : Entity 'mdash' not defined
	  compose a glyph yourself out of combining marks&mdash;and,
	                                                        ^
../usermanual-what-is-harfbuzz.xml:375: parser error : Opening and ending tag mismatch: ulink line 374 and a
	  url="http://fribidi.org/">fribidi</a> can do this for you.
	                                       ^
../usermanual-what-is-harfbuzz.xml:416: parser error : Entity 'mdash' not defined
      that you'll need&mdash;what you then do with the glyphs that your font
                             ^
../usermanual-what-is-harfbuzz.xml:429: parser error : Entity 'mdash' not defined
      shaping engine for OpenType fonts&mdash;&quot;HarfBuzz&quot; is
                                              ^
../harfbuzz-docs.xml:36: element include: XInclude error : could not load ../usermanual-what-is-harfbuzz.xml, and no fallback was found
../usermanual-install-harfbuzz.xml:57: parser error : Entity 'ndash' not defined
      On Fedora, RHEL, CentOS, or other Red-Hat&ndash;based systems, you would r
                                                      ^
../harfbuzz-docs.xml:37: element include: XInclude error : could not load ../usermanual-install-harfbuzz.xml, and no fallback was found
../usermanual-shaping-concepts.xml:42: parser error : Entity 'mdash' not defined
      operations&mdash;such as context-dependent substitutions,
                       ^
../harfbuzz-docs.xml:39: element include: XInclude error : could not load ../usermanual-shaping-concepts.xml, and no fallback was found
../xml/hb-unicode.xml:773: element refsect2: validity error : ID hb-unicode-funcs-t already defined
<refsect2 id="hb-unicode-funcs-t" role="typedef">
                                                ^
../xml/hb-buffer.xml:1802: element refsect2: validity error : ID hb-buffer-t already defined
<refsect2 id="hb-buffer-t" role="typedef">
                                         ^
../xml/hb-blob.xml:479: element refsect2: validity error : ID hb-blob-t already defined
<refsect2 id="hb-blob-t" role="typedef">
                                       ^
../xml/hb-face.xml:606: element refsect2: validity error : ID hb-face-t already defined
<refsect2 id="hb-face-t" role="typedef">
                                       ^
../xml/hb-font.xml:2054: element refsect2: validity error : ID hb-font-funcs-t already defined
<refsect2 id="hb-font-funcs-t" role="typedef">
                                             ^
../xml/hb-font.xml:2105: element refsect2: validity error : ID hb-font-t already defined
<refsect2 id="hb-font-t" role="typedef">
                                       ^
../xml/hb-set.xml:746: element refsect2: validity error : ID hb-set-t already defined
<refsect2 id="hb-set-t" role="typedef">
                                      ^
../xml/hb-shape-plan.xml:304: element refsect2: validity error : ID hb-shape-plan-t already defined
<refsect2 id="hb-shape-plan-t" role="typedef">
                                             ^
```

**Comment by n8willis** [comment]
Hmm. I guess I need to look for a better validator.

File: docs/usermanual-what-is-harfbuzz.xml (hunk 7 above)
Comment (n8willis) [reply docs/usermanual-what-is-harfbuzz.xml]: I don't follow. This seems perfectly straightforward; if you're not implementing a layout engine, then you rely on some dependency to do text shaping for you.

File: docs/usermanual-what-is-harfbuzz.xml (hunk 6 above)
Comment (n8willis) [reply docs/usermanual-what-is-harfbuzz.xml]: Well, this particular quoted text predates the 2.0.0 release, but I think that the new release illustrates that lack-of-CoreText does not mean lack-of-AAT is unavoidable.

File: docs/harfbuzz-docs.xml (hunk 3 above)
[the remark this answers is not in the document]
Comment (n8willis) [reply docs/harfbuzz-docs.xml]: Think this is fixed now; all DocBook parts need to have the DTD reference block or else they trigger this stuff.

File: docs/Makefile.am [hunk 8]
```diff
@@ -77,7 +77,7 @@ content_files=	\
 	usermanual-clusters.xml \
 	usermanual-fonts-and-faces.xml \
 	usermanual-glyph-information.xml \
-	usermanual-hello-harfbuzz.xml \
+	usermanual-getting-started.xml \
 	usermanual-install-harfbuzz.xml \
```
[the remark this answers is not in the document]
Comment (n8willis) [reply docs/Makefile.am]: OK, looks good.

**Comment by n8willis** [comment]
@behdad All checks are passing now; should be mergeable.

**Pull request merged**